packages/pigeon/lib/kotlin_generator.dart

```diff
@@ -241,7 +241,11 @@ class KotlinGenerator extends StructuredGenerator<KotlinOptions> {
     bool private = false,
   }) {
     final String privateString = private ? 'private ' : '';
-    final String classType = classDefinition.isSealed ? 'sealed' : 'data';
+    final String classType = classDefinition.isSealed
+        ? 'sealed'
+        : classDefinition.fields.isNotEmpty
+            ? 'data'
+            : '';
```

Comment:
Are empty data classes not supported on kotlin?

Reply:
Yeah

![image](https://github.com/user-attachments/assets/c13b5455-be23-40ac-bcc9-7c789433d955)